storage/oasis/nodeapi/api.go

```diff
@@ -206,10 +206,54 @@ type Proposal governance.Proposal
 // ....................  Runtimes  ....................
 // ....................................................
 
+// FallibleResponse stores the response to a node query. If the call was successful,
+// the response is stored in `result`. If the call failed due to a deterministic
+// error, eg call reverted, the error is stored in `deterministicErr`. This struct
+// should _not_ store transient errors or any other errors not considered to be a
+// valid node response.
+type FallibleResponse[Response any] struct {
+	Result           Response
```

Comment:
```suggestion
	Ok           *Response
```

Suggestions:
- I find `Result` confusing because it's what Rust uses for `FallibleResponse`.
- Add pointer to allow clearly expressing a missing ok-response.

Also, nit: I find it slightly odd to talk about deterministic errors in the docstring here; I'd probably put it in the docstring of `EVMSimulateCall`, because 1) that's where the semantics of this object are ultimately determined and 2) it can be confusing to consumers to have a return value `(FallibleResponse, error)` that can express errors in two different ways, without an explanation. OTOH the explanation is here, if they only dig a tiny bit. Your call.

Reply:
Updated both, and moved the comment as suggested